applications/serial_lte_modem/src/slm_at_host.c

```diff
@@ -705,6 +705,14 @@ void slm_at_receive(const uint8_t *buf, size_t len)
 		len -= ret;
 	}
 
+#if defined(CONFIG_SLM_UART_ECHO)
+	if (get_slm_mode() == SLM_AT_COMMAND_MODE && at_backend.send) {
+		const uint8_t clear_line_cmd[] = "\33[2K\r"; /* VT100 Clear line escape sequence */
+		at_backend.send(clear_line_cmd, sizeof(clear_line_cmd));
+		at_backend.send(slm_at_buf, at_cmd_len);
```

Comment:
This does not work with multi-line inputs such as certificates with AT%CMNG commands. Only the last line is echoed for the terminal. Similarly, if the certificate is larger than UART buffers, the received parts will be repeated multiple times. 

I also had problems with other multi-line inputs such as:
at#xsend="test
test"

The output buffer keeps on getting printed for every character on the second row.

Reply:
I am starting to think that this might be something that is quite hard to get to work correctly for all inputs. The local echo should handle this on terminals, so I don't fully understand the need for this. I am currently leaning on not approving this change to our main, even if it seems to work with most inputs.

Reply:
I've fixed multi-line input problem with my last push.
I implemented this to not only for terminal usage but also if someone writes an application to talk with modem through uart and expects echo on the first response line for the parser function etc. Because Modem Shell (mosh) has it by default true.
First, I created a ticket on Nordic DevZone: https://devzone.nordicsemi.com/f/nordic-q-a/116126/serial-lte-modem-doesn-t-show-what-i-typed
And then I realized there is no such an option for this. That's the why I wanted to implement my own. And I implemented with by default false. Therefore, it won't affect the other people who use SLM.
By the way, I already arranged my parser. I don't need this change. Even I need it I can patch it. So, I don't need this PR to be merged personally. It may be helpful for other people in the future.

Reply:
It is good that you got this to work in your parser. That is the way want to encourage our users to solve this issue. As for the integrations, ones that we currently have do not except echoes of commands. This need has not been risen during the time that I have worked with this component. I think that it may be more difficult to implement the parsing of responses if the original commands are echoed back.

If such need arises in future, this PR can be revisited.

For future reference, PR has the before mentioned issues: 

- Multi-line inputs keep repeating previous rows: 
at#xsend="test
test"

- Messages that are larger than the size of the UART buffers will be repeated multiple times.

There are sure to be other issues as well, but this will likely work well enough for some use-cases.